src/main/java/frc/robot/commands/ArmForeward.java

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Arm;
+
+public class ArmForeward extends CommandBase {
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+  private final Arm m_subsystem;
+
+  private boolean atFront;
+
+  public ArmForeward(Arm subsystem) {
+    m_subsystem = subsystem;
+    System.out.println("Ignore a warning about not using m_subsystem:" + m_subsystem);
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(subsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_subsystem.setMotorForeward();
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    m_subsystem.checkLimits();
+    atFront = m_subsystem.queryFront();
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    m_subsystem.stopMotor();
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
```

Comment:
You can jusut shorten this function to:

```java
public boolean isFinished(() {
    return m_arm.queryFront()
}
```

And it mightt then be useful/more clear to just name that method 'isAtFront'